### PR TITLE
feat: Marketplace v2

### PR DESCRIPTION
**Pull request by omniwired**

### CODE REVIEW

**Comment by WaDadidou** [comment]
This button is cropped, maybe the Collection/Activity container needs dynamic width. And add horizontal space to this container.
image
image

**Comment by WaDadidou** [comment]
Lonely Separators here, maybe you can add them in the JSX conditions to avoid display them anytime
image

**Comment by WaDadidou** [comment]
The price filter seems not working, I filter 1 to 20 but I see all collections

**Comment by WaDadidou** [comment]
Can you add a small horizontal space between the icon and the price ? (cart item)
image

**Comment by WaDadidou** [comment]
The gray in the selected collection is vertically shifted.
image
image

**Comment by WaDadidou** [comment]
Can you adjust the cart's head to fit with the Figma ?
- Full width separator (Also wfull width separators in the rest of the cart)
- Correct badge height
- Correct items number color
- Correct spaces
image
image

File: packages/screens/Marketplace/SideCart.tsx [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+  const dispatch = useAppDispatch();
+
+  const selectedNFTData = useSelector(selectAllSelectedNFTData);
+  const { setToastError } = useFeedbacks();
+
+  const balances = useBalances(wallet?.networkId, wallet?.address);
+  const hasEnoughMoney = selectedNFTData.every((nft) => {
+    const balance =
+      balances.find((bal) => bal.denom === nft.denom)?.amount || "0";
+    return balance > nft.price;
```
[the remark this answers is not in the document]
Comment (omniwired) [reply packages/screens/Marketplace/SideCart.tsx:229]: done

**Comment by omniwired** [comment]
> Can you adjust the filters's head to fit with the Figma ? image image

I feel them to be more consistent with the rest of the site now, unlike in the figma.
Let's discuss, but for now, is a "won't fix".

**Comment by omniwired** [comment]
> The gray in the selected collection is vertically shifted. image image

fixed

**Comment by WaDadidou** [comment]
> > Can you adjust the filters's head to fit with the Figma ? image image
> 
> I feel them to be more consistent with the rest of the site now, unlike in the figma. Let's discuss, but for now, is a "won't fix".

I think we need to follow the Figma for this point, since we have other elements that uses this kind of horizontal lines as "header", "footer" or "section" separators, especially in cards. But it's a detail (I'm on UI nazi mode)
Ex : 
image
image

**Comment by WaDadidou** [comment]
Could you add a loader to the "Buy now" button in the cart, or make this modal appears when the user click on "Buy now" in the cart ?
image

File: .env [hunk 2]
```diff
@@ -31,3 +31,6 @@ PRICES_SERVICE_URI=prices.mainnet.teritori.com:443
 DISPLAYED_NETWORKS_IDS=teritori,teritori-testnet,ethereum,ethereum-goerli,cosmos-hub,juno,osmosis,osmosis-testnet,solana,gno-testnet
 SELECTABLE_NETWORKS_IDS=teritori,teritori-testnet,ethereum,ethereum-goerli,osmosis,osmosis-testnet,gno-testnet
 IMG_PROXY_URL=https://imgproxy.tools.teritori.com/insecure/
+
+AIRTABLE_API_KEY_DAPPS_STORE=patZYqro8XEjo8qXT.97183a7a2a21decc37eeafe8da99f669876df0df109bc908c6311f4302c1991a
```
Comment (hthieu1110) [inline .env:35]: Is this safe to put this key here ? it's only for testing ?

File: go/pkg/marketplace/marketplace_service.go [hunk 3]
```diff
@@ -73,11 +74,18 @@ type DBCollectionWithExtra struct {
...
+	TotalVolume         float32
+	FloorPrice          uint64
+	MintPrice           string
+	NumTrades           int64
+	NumOwners           int32
+	Denom               string
 	MintContractAddress string
 	CreatorAddress      string
-	Price               uint64
+	Price               string
```
Comment (hthieu1110) [inline go/pkg/marketplace/marketplace_service.go:85]: Should double-check if this Price is used somewhere for calculation (front/back), and the change from uint64 => string does not break that calculation

File: go/pkg/marketplace/marketplace_service.go [hunk 4]
```diff
@@ -169,20 +179,49 @@ func (s *MarkteplaceService) Collections(req *marketplacepb.CollectionsRequest,
 				SELECT * FROM trades t2 INNER JOIN activities_on_period aop
 				ON aop.id = t2.activity_id
 			),
+     activities_on_period_comparision AS (
```
Comment (hthieu1110) [inline go/pkg/marketplace/marketplace_service.go:182]: Did you test the performance of this query ? I think for now, when we do not find another optimized way for this query, just be sure that of related fields are indexed

File: packages/screens/Marketplace/types.ts [hunk 5]
```diff
@@ -6,3 +6,9 @@ export type NFTActivityType = {
   buyer: string;
   seller: string;
 };
+
+export interface prettyPrint {
```
Comment (hthieu1110) [inline packages/screens/Marketplace/types.ts:10]: Should we Capitalize this => PrettyPrint ?

File: .env (hunk 2 above)
Comment (omniwired) [reply .env:35]: 🙈

File: packages/components/nfts/NFTView.tsx [hunk 6]
```diff
@@ -50,6 +67,17 @@ export const NFTView: React.FC<{
     useDropdowns();
   const [isTransferNFTVisible, setIsTransferNFTVisible] =
     useState<boolean>(false);
+  const localSelected = new Set(useSelector(selectSelectedNFTIds)).has(nft.id);
+  const dispatch = useAppDispatch();
+  const handleClick = (nft: NFT, selected: boolean) => {
+    if (width < 500) return; // disable cart on mobile
```
[the remark this answers is not in the document]
Comment (omniwired) [reply packages/components/nfts/NFTView.tsx:73]: is not enough the break provided by isMobile, this is a different value.

File: go/pkg/marketplace/marketplace_service.go (hunk 3 above)
Comment (omniwired) [reply go/pkg/marketplace/marketplace_service.go:85]: I've checked and I didn't find any issues.

File: packages/hooks/useMaxResolution.ts [hunk 7]
```diff
@@ -40,7 +40,8 @@ export const useMaxResolution = ({
   }, [windowWidth, isSidebarExpanded, noMargin, responsive, isMobile]);
 
   return {
-    width: width > screenContentMaxWidth ? screenContentMaxWidth : width,
+    width:
+      width > screenContentMaxWidthLarge ? screenContentMaxWidthLarge : width,
```
Comment (WaDadidou) [inline packages/hooks/useMaxResolution.ts:44]: Please, let `screenContentMaxWidth` here and add a property `isLarge` or something to `useMaxResolution` if you want to use `screenContentMaxWidthLarge`.
We need 1092px as max width for many screens to respect the design.
This hook starts to look weird, but we need to be careful with it for now.
image

**Review by WaDadidou — CHANGES_REQUESTED** [review]

**Review by WaDadidou — APPROVED** [review]

**Review by hthieu1110 — APPROVED** [review]
LGTM

**Review by WaDadidou — DISMISSED** [review]

**Review by hthieu1110 — DISMISSED** [review]
LGTM

**Review by hthieu1110 — APPROVED** [review]
GO :)

**Comment by zxxma** [comment]
LGTM.

**Pull request merged**